test/lib/heuristics/periodic_functions_test.rb

```diff
@@ -166,7 +166,9 @@ def test_add_missing_visits
       vrp = TestHelper.load_vrp(self, fixture_file: 'periodic_with_post_process')
       vrp.vehicles = TestHelper.expand_vehicles(vrp)
       s = Wrappers::PeriodicHeuristic.new(vrp)
-      s.instance_variable_set(:@candidate_routes, Marshal.load(File.binread('test/fixtures/add_missing_visits_candidate_routes.bindump'))) # rubocop: disable Security/MarshalLoad
+      add_missing_visits_candidate_routes = Marshal.load(File.binread('test/fixtures/add_missing_visits_candidate_routes.bindump')) # rubocop: disable Security/MarshalLoad
+      add_missing_visits_candidate_routes.each_value{ |vehicle| vehicle.each_value{ |route| route[:matrix_id] = vrp.vehicles.first.matrix_id } }
+      s.instance_variable_set(:@candidate_routes, add_missing_visits_candidate_routes)
```

Comment:
Why do you want to re-dump a binary file?

This correction is not because the binary file is "incorrect" _per se_ but it is because of the fact that now the matrices gets their id automatically inside the api (by active_hash), so modifying the dump doesn't prevent an id mismatch. However, actually using the "correct" `matrix_id`s coming from the `vrp.vehicle.matrix_id` does. Note that this modification mimics the code -- i.e., this variable would arrive periodic visit with `route[:matrix_id] = vrp.vehicles.first.matrix_id` if we were to launch the optim via `wrapper_vrp`.

Reply:
I know that, I meant redumping the file with the changes you wrote in order to avoid doing these operations everytime we run the test. Since this dump file is specific for this test I think we can just adjust dump file according with our new behaviour.

Reply:
I clearly understood that you mean  "redumping the file with these changes " but like I explained, the id's are auto generated by active_hash now and this is the way to handle that. Otherwise, we need to redump megabytes of data to the git history each time something changes the id generation order. This operation literally takes nanoseconds and prevent all future false negatives.